main.cc

```diff
@@ -1071,7 +1071,7 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
                 reinterpret_cast<service::storage_proxy_stats::stats*>(ptr)->register_stats();
                 reinterpret_cast<service::storage_proxy_stats::stats*>(ptr)->register_split_metrics_local();
             };
-            proxy.start(std::ref(db), std::ref(gossiper), spcfg, std::ref(node_backlog),
+            proxy.start(std::ref(db), std::ref(gossiper), std::ref(ss), spcfg, std::ref(node_backlog),
```

Comment:
/cc @xemul is this taking us into a dependency loop?

re the name of the partitioner, you can just pass it into query_processor during its construction. It's static and never changes. You can even get it from db::config.

Do you need storage_service for ownership map? I'm sure storage_proxy has access to it.

Reply:
> /cc @xemul is this taking us into a dependency loop?

It is. In at least three:
- proxy -> storage-service -> migration-manager -> proxy
- proxy -> storage-service -> repair -> proxy
- proxy -> storage-service -> batchlog -> query-processor -> proxy

Reply:
The storage_service is only needed to call .describe_ring() on. This method, in turn, only needs effective_replication_map instance at hand, so it doesn't even need to be the storage_service method

Reply:
I've tried to extract the describe_ring() function to be able to call it both from storage_service and storage_proxy but it turned out it has more dependencies in the helper methods (I belive it was gossip and something else). So I've created noncopyable_function that does describe_ring() and I've moved it to the storage_service.

I've also passed the partitioner name to the storage_proxy constructor and only now I did see it can be obtained from db::config. It's quite late right now so next day I'll remove it from storage service entirely and I'll take it from db::config.

Reply:
Removed partitioner name from storage_proxy and using the one from db::config